parastell/invessel_build.py

```diff
@@ -1281,9 +1316,13 @@ def create_mesh(self):
                 continue  # Skip iteration if a gap is indicated
             for toroidal_idx in range(self._num_ribs - 1):
                 for poloidal_idx in range(self._num_rib_pts - 1):
-                    self._create_tets_from_hex(
+                    tets, vertex_id_list = self._create_tets_from_hex(
                         surface_idx, poloidal_idx, toroidal_idx
                     )
+                    [
+                        self._compute_tet_data(tet_ids, tet)
+                        for tet_ids, tet in zip(vertex_id_list, tets)
+                    ]
```

Comment:
If `_compute_tet_data` returned the `tet_vol` you could `self.volumes.extend()` by the list you create here.